subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -648,14 +655,14 @@ static void isr_rx(void *param)
 		}
 
 		/* Get reference to stream context */
-		stream_handle = lll->stream_handle[lll->stream_curr];
+		stream_handle = lll->stream_handle[stream_curr];
```

Comment:
`stream_curr` is a generic and undocumented field, so it's hard to understand what it is. 

Is it an index starting from 0 to `num_bis`? 

How does it relate to the other (undocumented) fields?

- Is `num_bis` the number of BIS currently synced to, or the number of BIS in the BIG? 
- What is the relationsship between `num_bis` and `stream_count`?
- How does it related to `bis_curr`? 
- Somewhat related, is `bis_idx` related to the BIS indexes (starting from 0x01), or is another index for BIS unrelated to the BIS indexes? 

In the changes below you have a check `if ((stream_curr < BT_CTLR_SYNC_ISO_STREAM_MAX) &&` which indicitates the `stream_curr` can sometimes be larger than `BT_CTLR_SYNC_ISO_STREAM_MAX`, but that check is only there any not anywhere where you use it directly as an index to `stream_handle`.

Reply:
> Is it an index starting from 0 to num_bis?

`stream_curr` hold 0 indexed stream that is currently synchronized to. It ranges upto the number of streams required to be sync-ed. i.e. `0 <= stream_curr < stream_count`  where `0 < stream_count <= num_bis`.

> Is num_bis the number of BIS currently synced to, or the number of BIS in the BIG?

the number of BIS in the BIG

> What is the relationsship between num_bis and stream_count?

`0 < stream_count <= num_bis`

> How does it related to bis_curr?

`bis_curr` is the currently synchronized BIS amongst the total BIS from the Broadcaster.

> Somewhat related, is bis_idx related to the BIS indexes (starting from 0x01), or is another index for BIS unrelated to the BIS indexes?

yes, `bis_idx` is the BIS indexes (starting from 0x01) and local-variable corresponding to `bis_curr` in the LLL context structure.


> In the changes below you have a check if ((stream_curr < BT_CTLR_SYNC_ISO_STREAM_MAX) && which indicitates the stream_curr can sometimes be larger than BT_CTLR_SYNC_ISO_STREAM_MAX, but that check is only there any not anywhere where you use it directly as an index to stream_handle.

The check is added to handle compiler warning that the `payload` array could be accessed out-of-bound using `stream_curr` (because the compiler thinks `stream_count` could have greater value than `BT_CTLR_SYNC_ISO_STREAM_MAX`